client/src/providers/TabContext.js

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useReducer } from 'react';
+import { tabColumns } from '../constants';
+
+const TabContext = React.createContext();
```

Comment:
TabContext is very generic, I will suggest a specific name like SiteDetailViewTabContext / SiteDetailTabContext